acceptance/lib/pxp-agent/config_helper.rb

```diff
@@ -9,6 +9,18 @@
 
 PCP_VERSION = ENV['PCP_VERSION'] || '2'
 
+HOCON_RENDER_OPTIONS = Hocon::ConfigRenderOptions.new(
```

Comment:
I thought we had a helper around this, but haven't found it yet. @ekinanp or @zreichert do you remember any?

Reply:
The only ones I'm aware of are what's here: https://github.com/puppetlabs/beaker/blob/master/lib/beaker/dsl/helpers/hocon_helpers.rb (what you pointed me to when doing PCP-771).

Reply:
That's what I was thinking of, thanks!

Reply:
We already use those in the task_downloads test, we should use them here as well.

Reply:
Unfortunately the helpers in the referenced file don't cater for the use case in these tests well as they assume modifying an existing file. Attempts to replace the entire content of the file lead to clumsy code at best:
- the simplest I could come up with was clearing the file first and then adding the individual config values in a loop:
  ```ruby
  config = {
    :a => 1,
    :b => 2
  }

  # clear the remote file first
  create_remote_file(host, file, '')

  hocon_file_edit_in_place_on(host, file) do |host, doc|
    # insert the config values into the ruby-hocon config document object
    config.reduce(doc) { |doc, (k, v)|
      doc.set_config_value(k.to_s, Hocon::ConfigValueFactory.from_any_ref(v)) 
    }
  end
  ```
  but that approach involves three round trips to the host:
  1. clearing the file
  2. reading the file in a helper called from `hocon_file_edit_in_place_on`
  3. writing the file in `hocon_file_edit_in_place_on`

- I was able to eliminate the clearing of the file by first generating the config in the hocon format in exactly the same way I propose in this PR and then parsing it into the ruby-hocon config document which is expected as the return value from the block passed to the `hocon_file_edit_in_place_on` helper:
  ```ruby
  HOCON_RENDER_OPTIONS = Hocon::ConfigRenderOptions.new(
    false,  # origin_comments?
    true,   # comments?
    true,   # formatted?
    false,  # json?
    :colon, # key_value_separator (:colon or :equals)
  )

  config = {
    :a => 1,
    :b => 2
  }

  hocon_file_edit_in_place_on(host, file) do |host, doc|
    # searialize the config first
    serialized_doc = Hocon::ConfigValueFactory.from_any_ref(config).render(HOCON_RENDER_OPTIONS)
    # now parse it into the ruby-hocon config document object
    Hocon::Parser::ConfigDocumentFactory.parse_string(serialized_doc)
  end
  ```
   but that still involves two round trips and besides it's just a more complicated way of doing the same I propose in this PR

To sum it up: I don't think using the current helpers is a reasonable approach to take, but I'll be happy to add a new helper which would cater for our use case better and use that.
What would be your suggestion @MikaelSmith?

Reply:
Actually it is necessary to create the file even in the second case, as if the file doesn't exist at the time the `hocon_file_edit_in_place_on` helper is called, an exception is thrown.